models/cart-discount/src/cart-discount-line-items-target/cart-discount-line-items-target-draft/transformers.ts

```diff
@@ -13,11 +13,15 @@ const transformers = {
     TCartDiscountLineItemsTargetDraft,
     TCartDiscountLineItemsTargetDraftGraphql
   >('graphql', {
-    replaceFields: ({ fields }) => ({
-      lineItems: {
-        ...fields,
-      },
-    }),
+    replaceFields: ({ fields }) => {
+      const { type, ...rest } = fields;
+
+      return {
+        [type]: {
+          ...rest,
+        },
+      };
+    },
```

Comment:
You'll see this pattern throughout. GraphQL doesn't have a `type` field colocated with the data like the HTTP API. For `target` and `value` fields for both Cart and Product Discount drafts, we have to transform the shape.

ex. HTTP API
```
{
  type: 'foo'
  otherData: 'here'
  somethingElse: 'there'
}
```
ex. GraphQL API
```
{
  foo: {
    otherData: 'here'
    somethingElse: 'there'
  }
}
```

Docs:
![image](https://github.com/commercetools/test-data/assets/15024562/53a9a30d-a3a9-46ed-a1cd-338d63672f4f)